controller/linodeobjectstoragebucket_controller_test.go

```diff
@@ -0,0 +1,183 @@
+// /*
+// Copyright 2023 Akamai Technologies, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/linode/linodego"
+
+	infrav1 "github.com/linode/cluster-api-provider-linode/api/v1alpha1"
+	"github.com/linode/cluster-api-provider-linode/cloud/scope"
+	"github.com/linode/cluster-api-provider-linode/mock"
+	"github.com/linode/cluster-api-provider-linode/util"
+)
+
+var _ = Describe("LinodeObjectStorageBucket controller", func() {
+	ctx := context.Background()
+
+	obj := &infrav1.LinodeObjectStorageBucket{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "sample-bucket",
+			Namespace: "default",
+		},
+		Spec: infrav1.LinodeObjectStorageBucketSpec{
+			Cluster: "cluster",
+			Label:   util.Pointer("sample"),
+		},
+	}
+
+	recorder := record.NewFakeRecorder(3)
+
+	secretName := fmt.Sprintf(scope.AccessKeyNameTemplate, *obj.Spec.Label)
+
+	var secret corev1.Secret
+	var mockCtrl *gomock.Controller
+
+	BeforeEach(func() {
+		// Create a new gomock controller for each test run
+		mockCtrl = gomock.NewController(GinkgoT())
```

Comment:
I'm curious: why create a new ctrl before each run? Why not just create it once and use it?

Reply:
It's needed because each gomock controller has its own scope. Invoking `mockctrl.Finish` asserts all of the declared mock expectations for its particular instance.